### ISSUE TITLE
Creating encoded videos without javascript

### ISSUE THREAD

**Issue by clvcooke** [w3c/webcodecs#194]
As part my use-case for webcodecs I'm hoping to batch-process my videos to generate the encoded chunks first (and serve them in an unconventional manner, hence the need for webcodecs). 

I'm able to run the demo provided at: https://webcodecs-blogpost-demo.glitch.me/ . Additionally, I have modified it to use my own data instead of the sample data. 

**My question is:** how do I generate the _encoded video chunks_ in a format appropriate for the _video decoder_
 without using the provided _video encoder_ ? I haven't been able to find a spec for how the encoded chunks are generated, so I'm having a hard time matching the formatting with a non JS tool (for example ffmpeg). I'm flexible in tooling, but would it would be ideal to be able to do this without needing to run client-side javascript.

**Comment by sandersdan** [comment]
The most reliable way to generate chunks is to mux as an MP4, then extract each sample (one sample = one chunk). Note that for H.264 this will produce data in AVC format, so you'll need to also extract the avcC and provide it in `VideoDecoderConfig.description`.

We are planing to provide a basic level of format documentation in the WebCodecs codec registry (https://w3c.github.io/webcodecs/codec_registry.html), but that resource will still assume a high level of familiarity with bitstreams and containers.

In general I would recommend serving MP4 and using an MP4 demuxing library in JS to generate chunks. We are planning to create some example code in the near future.

**Comment by clvcooke** [comment]
Just as a quick follow-up, I was able to get things working by just parsing up a webm file. 

I figured it out through trial and error, but the information that I was looking for was that the encoded chunks just need the *data* component of the webm frame. Since each chunk in a VP9 encoded file has a header and data, I was able to just ignore the header, and split the .webm video file (encoded with ffmpeg) into chunks more or less manually.

For example, for a single chunk within the webm file, you would see something like this:
![image](https://user-images.githubusercontent.com/7168476/116601141-1e619700-a8f8-11eb-8dd4-4af21d3bc617.png)

To generate data which is compatible with the VideoDecoder you just need to take the part after the header (the non-colored portion). 

I realize that my application may be atypical, and my background is a little weird (I don't normally work with video encoders/decoders), but hopefully other folks find this useful.
